lib/Onyx.js

```diff
@@ -909,6 +909,21 @@ function hasPendingMergeForKey(key) {
     return Boolean(mergeQueue[key]);
 }
 
+/**
+ * For objects, the key for null values needs to be removed from the object to ensure the value will get removed from storage completely.
+ * On native, SQLite will remove top-level keys that are null. To be consistent, we remove them on web too.
+ * @private
+ * @param {*} value
+ * @returns {*}
+ */
+function cleanObject(value) {
+    if (!_.isArray(value) && _.isObject(value)) {
+        return _.omit(value, objectValue => _.isNull(objectValue));
+    }
+
+    return value;
```

Comment:
Early return (suggestion is untested)
```suggestion
    if (_.isArray(value) || !_.isObject(value)) {
        return value;
    }

    return _.omit(value, objectValue => _.isNull(objectValue));
```